### PR TITLE
AboutBase section

### PR DESCRIPTION
**Pull request by veronika-kolosova**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: src/pages/AboutBase/AboutBase.tsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+    const handleEdit = (title: string, id: SectionModel['id']) =>{
+      setData(prev=>prev.map(item=>item.id===id?{...item,title}:item))
+    }
+
+    const handleAdd = async (index:any) => {
+      setLoading(false);
+      const newItem = {id:count++, title:'', subsection: JSON.parse(index.subsection)}
+      // const res: SectionModel[] = await Api.get();
+      // setData(res);
```
Comment (OleksandrBilan) [inline src/pages/AboutBase/AboutBase.tsx:67]: Do you need these comments?

File: src/pages/AboutBase/AboutBase.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    const handleSubmit = async(values:any)=>{
+      const newSection: SectionModel = {
+        id:0,
+        title:"",
+        subsection:JSON.parse(values.subsection)
+      }
+    }
+    const editOtlined = () => (
+      <EditOutlined className="editInfoIcon"
+        //onClick={() =>history.push(``)}
```
Comment (OleksandrBilan) [inline src/pages/AboutBase/AboutBase.tsx:81]: and this

File: src/pages/AboutBase/AboutBase.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+          // If you don't want click extra trigger collapse, you can prevent this:
+          event.stopPropagation();
+        }}
+      />
+    );
+
+    const deleteOtlined = () =>(
+    <DeleteOutlined className="deleteInfoIcon"
+        //onClick={() => seeDeleteModal()}
+        //onClick={()=>handleDelete(item.id)}
```
Comment (OleksandrBilan) [inline src/pages/AboutBase/AboutBase.tsx:92]: and here

File: src/pages/AboutBase/AboutBase.tsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        }}
+    />
+    );
+    const showModal = () => setVisibleModal(true);
+    
+    return !loading ? (
+        <Layout.Content className="aboutbase">
+            <Title level={1}>Про базу</Title>
+            <div className="searchContainer">
+            {
```
Comment (OleksandrBilan) [inline src/pages/AboutBase/AboutBase.tsx:105]: Why do you place here the '{' ??

File: src/pages/AboutBase/AboutBase.tsx [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+                <Button type="primary" onClick={showModal}>
+                  Задати запитання
+                </Button>
+            }
```
Comment (OleksandrBilan) [inline src/pages/AboutBase/AboutBase.tsx:109]: and here, i think you dont need these

File: src/pages/AboutBase/AboutBase.tsx [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+            <Search
+          placeholder="Пошук"
+          enterButton
+          onChange={handleSearch}
+        />
+            </div>
+    {Sections.map((item,index)=>(
+    <Collapse
```
Comment (OleksandrBilan) [inline src/pages/AboutBase/AboutBase.tsx:117]: maybe some tabulation here?

File: src/pages/AboutBase/AboutBase.tsx [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+  </Collapse>
+    ))}
+    <div className="addSection">
+  <Input placeholder=" Додати розділ"/>
+  <Button type="primary" onClick={()=>handleAdd}>?</Button>
+  </div>
+  <AskQuestionModal
+            setVisibleModal={setVisibleModal}
+            visibleModal={visibleModal}
+            //onAdd={handleAdd}
```
Comment (OleksandrBilan) [inline src/pages/AboutBase/AboutBase.tsx:137]: apply the correct tabulation, please

File: src/pages/AboutBase/AskQuestionModal.tsx [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    <Drawer
+      width="auto"
+      title="Задати запитання"
+      visible={visibleModal}
+      onClose={handleCancel}
+      footer={null}
+    >
+      <FormAskQuestion  
+        setVisibleModal = {setVisibleModal}
+         />
```
Comment (OleksandrBilan) [inline src/pages/AboutBase/AskQuestionModal.tsx:26]: tabulation

File: src/pages/AboutBase/Subsections.tsx [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+} from "@ant-design/icons";
+import "./AboutBase.less";
+import "./Subsections.css";
+import SubsectionModel from '../../models/AboutBase/SubsectionModel';
+
+
+const {Content} = Layout;
+const { Panel } = Collapse;
+
+function callback(key: any) {
```
Comment (OleksandrBilan) [inline src/pages/AboutBase/Subsections.tsx:23]: what is the point of this function?

File: src/pages/AboutBase/Subsections.tsx [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+          id:0,
+          sectionId:0,
+          title:"",
+          description:""
+      }
+    ]);
+
+
+    const handleSubDelete=(id:SubsectionModel['id'])=>
+    {
```
Comment (OleksandrBilan) [inline src/pages/AboutBase/Subsections.tsx:44]: bracket on previous line

File: src/pages/AboutBase/Subsections.tsx [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+    const handleSubEdit = (title: string, id: SubsectionModel['id']) =>{
+      setData(prev=>prev.map(item=>item.id===id?{...item,title}:item))
+    }
+
+    const handleSubAdd = async (index:any) => {
+      setLoading(false);
+      //const newItem = {id:++, title:'', description:'', sectionId:''}
+      // const res: SectionModel[] = await Api.get();
+      // setData(res);
+      //setData(prev => [...prev.slice(0, index + 1), newItem, ...prev.slice(index + 1)])
```
Comment (OleksandrBilan) [inline src/pages/AboutBase/Subsections.tsx:57]: do something with the comments

File: src/pages/AboutBase/Subsections.tsx [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+      setLoading(true);
+    };
+
+    
+    
+    const editOtlined = () => (
+      <EditOutlined className="editInfoIcon"
+        //onClick={() =>history.push(``)}
```
Comment (OleksandrBilan) [inline src/pages/AboutBase/Subsections.tsx:65]: comments

File: src/pages/AboutBase/Subsections.tsx [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+                key="1"
+                extra={<>{editOtlined()}{deleteOtlined()}</>}
+            >
+
+            </Panel>
+    </Collapse>
+    </div>
+    <div className="modalDiv"> 
+    <Popup modal trigger={<Button className="addPostButton" type="primary" /*onClick={addPost}*/>Додати допис</Button>}>
+    <div><TextEditor /></div>
```
Comment (OleksandrBilan) [inline src/pages/AboutBase/Subsections.tsx:106]: tabulation

File: src/pages/AboutBase/TextEditor.tsx [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+                [{ 'align': [] }],
+                ['clean']
+    ]}
+    return (
+         <Form 
+         className="form" 
+         name="basic" 
+         onFinish={handleSubmit} 
+         layout="vertical"
+         form={form}>
```
Comment (OleksandrBilan) [inline src/pages/AboutBase/TextEditor.tsx:61]: tabulation

File: src/pages/AboutBase/TextEditor.tsx [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+          <Form.Item
+            className={formclasses.formField}
+            label="Опис"
+            labelCol={{ span: 24 }}
+            rules={[{ required: true}]}>
+            {/* <ReactQuill 
+            className={formclasses.text} 
+            theme="snow" value={value} 
+            onChange={setValue}
+            modules={modules}/> */}
```
Comment (OleksandrBilan) [inline src/pages/AboutBase/TextEditor.tsx:87]: remove comments

**Review by OleksandrBilan — CHANGES_REQUESTED** [review]

**Review by christykobyletska — APPROVED** [review]

**Review by sharix4 — APPROVED** [review]

**Review by nicoolle — APPROVED** [review]

**Review by solomiyon — APPROVED** [review]

**Review by OleksandrBilan — APPROVED** [review]

**Pull request merged**